rescript/ncbi.py

```diff
@@ -84,28 +89,34 @@ def _get(params, ids=None, entrez_delay=0.):
         r = requests.get(esearch, params=params)
         r.raise_for_status()
         webenv = parse(r.content)['eSearchResult']
+        if 'WebEnv' not in webenv:
+            raise ValueError('No sequences for given query')
         params = dict(
             db='nuccore', rettype='fasta', retmode='xml',
             WebEnv=webenv['WebEnv'], query_key=webenv['QueryKey']
         )
         expected_num_records = int(webenv['Count'])
 
     efetch = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi'
-    # Need to better handle the case of invalid ids
-    # Need to better handle intermittent fault where we
-    #  don't get the whole data set back
     data = []
     while len(data) < expected_num_records:
         params['retstart'] = len(data)
         time.sleep(entrez_delay)
         r = requests.get(efetch, params=params)
-        r.raise_for_status()
+        if r.status_code != requests.codes.ok:
+            content = parse(r.content)
+            content = list(content.values()).pop()
+            warnings.warn('Download did not finish smoothly. The following '
```

Comment:
This approach seemed better because it tells the user that something went wrong but also allows the user to check which accessions didn't come back.

We could make it better by raising a firmer warning, for instance the sort that shows up even without `--verbose` specified. Alternatively I could prepare a list of accessions that failed to download and raise an error with those in the message. That message could potentially be quite big, though.

Reply:
maybe some kind of log file would be appropriate? I think outputting a log file as text is fine in this case (there is precedent with deblur), unless if you think that failures should cause a fatal error. (it's not clear to me what would cause this failure)

Reply:
This error occurs when one of your accession ids is wrong or if something goes wrong on the ncbi end.

Reply:
well that's a shame if we can't differentiate the two errors... sounds like a log is our best bet.

Reply:
I decided to raise an error instead of a warning. I have added a more descriptive error message that will hopefully be useful in most cases.